base_layer/core/src/proof_of_work/difficulty.rs

```diff
@@ -118,6 +118,9 @@ pub mod util {
     /// This will provide the difficulty of the hash assuming the hash is big_endian
     pub(crate) fn big_endian_difficulty(hash: &[u8]) -> Difficulty {
         let scalar = U256::from_big_endian(hash); // Big endian so the hash has leading zeroes
+        if scalar <= U256::from(2).pow(U256::from(192)) {
+            return u64::MAX.into();
+        }
```

Comment:
It's simpler to do the U256 division in #124 first, and then check for overflow:

```
let scalar = U256::from_big_endian(hash);
let result = U256::MAX / scalar;
let result = result.min(u64::MAX.into())
```

Reply:
Thank you for the suggestion, I changed the code accordingly.